sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestBase.java

```diff
@@ -0,0 +1,243 @@
+            testProfile = PLAYBACK_PROFILE;
+
+            List<HttpPipelinePolicy> policies = new ArrayList<>();
+            policies.add(new TextReplacementPolicy(interceptorManager.getRecordedData(), textReplacementRules));
+            policies.add(new HostPolicy(PLAYBACK_URI + "/"));
+            policies.add(new CookiePolicy());
+            httpPipeline = buildHttpPipeline(null, testProfile, policies, interceptorManager.getPlaybackClient());
+
+            textReplacementRules.put(PLAYBACK_URI_BASE + "1234", PLAYBACK_URI);
+            addTextReplacementRules(textReplacementRules);
+            logger.info(PLAYBACK_URI);
+        } else {
+            if (System.getenv(AZURE_AUTH_LOCATION) != null) { // Record mode
+                final File credFile = new File(System.getenv(AZURE_AUTH_LOCATION));
+                try {
+                    testAuthFile = AuthFile.parse(credFile);
+                } catch (IOException e) {
+                    throw logger.logExceptionAsError(new RuntimeException("Cannot parse auth file. Please check file format."));
+                }
+                credential = testAuthFile.getCredential();
+                testProfile = new AzureProfile(testAuthFile.getTenantId(), testAuthFile.getSubscriptionId(), testAuthFile.getEnvironment());
+            } else {
+                Configuration configuration = Configuration.getGlobalConfiguration();
+                String clientId = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_ID);
+                String tenantId = configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID);
+                String clientSecret = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_SECRET);
+                String subscriptionId = configuration.get(Configuration.PROPERTY_AZURE_SUBSCRIPTION_ID);
+                if (clientId == null || tenantId == null || clientSecret == null || subscriptionId == null) {
+                    throw logger.logExceptionAsError(
+                        new IllegalArgumentException("When running tests in record mode either 'AZURE_AUTH_LOCATION' or 'AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET and AZURE_SUBSCRIPTION_ID' needs to be set"));
+                }
+
+                credential = new ClientSecretCredentialBuilder()
+                    .tenantId(tenantId)
+                    .clientId(clientId)
+                    .clientSecret(clientSecret)
+                    .authorityHost(AzureEnvironment.AZURE.getActiveDirectoryEndpoint())
+                    .build();
+                testProfile = new AzureProfile(tenantId, subscriptionId, AzureEnvironment.AZURE);
+            }
+
+            List<HttpPipelinePolicy> policies = new ArrayList<>();
+            policies.add(new TimeoutPolicy(Duration.ofMinutes(1)));
+            policies.add(new CookiePolicy());
+            if (!interceptorManager.isLiveMode()) {
+                policies.add(new TextReplacementPolicy(interceptorManager.getRecordedData(), textReplacementRules));
```

Comment:
TextReplacementPolicy would take action for replacement by the rules we enter into the map.